### PR TITLE
Blog-app: Add forms

### PR DESCRIPTION
**Pull request by GedeonTS**
## Blog-app: Add forms

In this project I:
-  Created a method called `current_user` in your ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
- Created forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` .
    - Created a comment on behalf of the `current_user` .
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/views/likes/_create.html.erb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<section>
+  <%= form_with model:@like do |f|%>
+      <%= f.submit "Like" , data: {disable_with: "Like"}, class: "add-btn like"%>
+  <%end%>
+</section>
```
Comment (FrederickMih) [inline app/views/likes/_create.html.erb]: - [ ] Hi @GedeonTS, based on the requirements of this project, users should be able to add a like to a post. When I click on `Like`, I got the below console error.

Screenshot from 2022-08-20 18-50-38


Kindly fix it.

File: config/routes.rb [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
 
 Rails.application.routes.draw do
-  get 'users', to: 'users#index'
-  get 'users/:user_id/posts', to: 'posts#index'
-  get 'users/:user_id/posts/:id', to: 'posts#show'
-  get 'users/:id', to: 'users#show'
+  root 'users#index'
+  resources :users,only:[:index, :show] do
+    resources :posts, only:[:index, :new, :create, :show] do
+    resources :comments, only: [:create]
```
Comment (FrederickMih) [inline config/routes.rb:7]: - [x] Based on the requirements of this project, they should be a pair of routes for creating comments, that's 

```ruby
POST   /posts/:post_id/comments(.:format)     comments#create
 GET   /posts/:post_id/comments/new(.:format)     comments#new
```

Note that you are missing the new action in your comments controllers and you should equally have a view for it.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @GedeonTS , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- GitHub workflow has been followed :heavy_check_mark: 
- PR has a title and summary :heavy_check_mark: 
- No Linter error detected :heavy_check_mark: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 2 above)
Comment (GedeonTS) [reply config/routes.rb:7]: Thanks @FrederickMih, let me do it right away!

**Review by t-yanick — APPROVED** [review]
Hi @GedeonTS, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by GedeonTS** [comment]
Thanks @t-yanick :ok_hand:

**Pull request merged**